x/proof/keeper/msg_server_create_claim_test.go

```diff
@@ -15,16 +17,129 @@ import (
 	"github.com/pokt-network/poktroll/x/proof/keeper"
 	"github.com/pokt-network/poktroll/x/proof/types"
 	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	"github.com/pokt-network/poktroll/x/shared"
 	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
 )
 
 var defaultMerkleRoot = []byte{0, 1, 0, 1}
 
 func TestMsgServer_CreateClaim_Success(t *testing.T) {
+	tests := []struct {
+		desc           string
+		getClaimHeight func(
+			ctx context.Context,
+			keepers *keepertest.ProofModuleKeepers,
+			sessionHeader *sessiontypes.SessionHeader,
+		) int64
+	}{
+		{
+			desc: "claim window open height",
+			getClaimHeight: func(
```

Comment:
Could we not do something like `getClaimHeight: shared.GetClaimWindowOpenHeight` which would simplify everything?

Reply:
Since these test cases all involve persistence, we would either have to construct fresh keepers for each scenario or add test code to revert the persistence side effects (i.e. delete claims). I opted for the format because the latter seemed like it may be more difficult to maintain.